test/e2e/common/lifecycle_hook.go

```diff
@@ -38,7 +40,7 @@ var _ = framework.KubeDescribe("Container Lifecycle Hook", func() {
 		preStopWaitTimeout   = 30 * time.Second
 	)
 	Context("when create a pod with lifecycle hook", func() {
-		var targetIP string
+		var targetIP, targetURL string
```

Comment:
I was hoping we can have a more consistent way to handle IPv6 cases like this --- just a bit worried to see subtle logic scattered around.

Though I guess this is okay for now?

Reply:
@MrHohn I can add a function to build correct IPv6 urls so it can be reused 

```
// FormatLiteralIPURL Returns a valid URL for literal IP addresses
// https://tools.ietf.org/html/rfc2732
func FormatLiteralIPURL (ip string) string {
			if strings.Contains(ip, ":") {
				return fmt.Sprintf("[%s]", ip)
			}
                        return ip
}
``` 

just let me know what will be the best place to add it

Reply:
Looks good, probably add to https://github.com/kubernetes/kubernetes/blob/master/test/e2e/framework/networking_utils.go for now. We can find a more common place outside of the test folder later if needed.